File: MosPosudit.Model/Requests/User/UserProfileUpdateRequest.cs
using System.ComponentModel.DataAnnotations;

namespace MosPosudit.Model.Requests.User
{
    public class UserProfileUpdateRequest
    {
        [StringLength(50)]
        public string? FirstName { get; set; }

        [StringLength(50)]
        public string? LastName { get; set; }

        [StringLength(50)]
        public string? Username { get; set; }

        [StringLength(100)]
        [EmailAddress(ErrorMessage = "Invalid email format")]
        public string? Email { get; set; }

        [StringLength(20)]
        [Phone]
        public string? PhoneNumber { get; set; }

        public byte[]? Picture { get; set; }
    }
} 

File: MosPosudit.UI/shared/lib/dtos/user/user_profile_update_request.dart
class UserProfileUpdateRequestDto {
  final String? firstName;
  final String? lastName;
  final String? username;
  final String? email;
  final String? phoneNumber;
  final String? picture;

  UserProfileUpdateRequestDto({
    this.firstName,
    this.lastName,
    this.username,
    this.email,
    this.phoneNumber,
    this.picture,
  });

  Map<String, dynamic> toJson() {
    final json = <String, dynamic>{};
    if (firstName != null) json['firstName'] = firstName;
    if (lastName != null) json['lastName'] = lastName;
    if (username != null) json['username'] = username;
    if (email != null) json['email'] = email;
    if (phoneNumber != null) json['phoneNumber'] = phoneNumber;
    if (picture != null) json['picture'] = picture;
    return json;
  }
}

